tests/cases/compiler/NonNullableInNonStrictMode.ts

```diff
@@ -0,0 +1,18 @@
+// @strict: false

```

Comment:
I think it would be good to actually test these behaviors in both strict on and off.

```suggestion
// @strict: true,false
```

Then rename the test to `nonNullablesAndObjectIntersections.ts`